### PR TITLE
ci: Update `pre-commit`

### PR DESCRIPTION
**Pull request by eddiebergman**
#### Reference Issues/PRs
None

#### What does this implement/fix? Explain your changes.
* `pre-commit autoupdate`
* Remove unneeded type dependanices for mypy in pre-commit (you don't directly use pyyaml or psutil)
* Remove `pyyaml` from `pyproject.toml` as you don't directly use it (it's a transitive dependancy) which should be specified by the dependency itself.
* Fix one issue the newer version of `ruff` picked up, i.e. `assert type(x) == np.ndarray => assert isinstance(x, np.ndarray)`

#### Minimal Example / How should this PR be tested?
`pre-commit run --all-files`

#### Any other comments?

---
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the
terms of your choice.

### CODE REVIEW

**Comment by eddiebergman** [comment]
@rheasukthanker I looked into it and the only difference was that `pre-commit` and `unit-test` were told that they should come from `Any Source` as opposed to `Github Actions`

I've updated the ruleset so it matches... Lets see if that works


Screenshot_2024-08-01_12-32-27

**Comment by eddiebergman** [comment]
@rheasukthanker fixed, the issue is that it was targeting `pre-commit` but it should have been targeting `run-all-files` as indicated here

Screenshot_2024-08-01_12-40-00

I find it weird that in the rule-sets it allows you to add things don't exist in your set of possible actions.

**Tldr** it was waiting on a non-existant thing to pass. I assume this rule-set was a new change
